set.mm

```diff
@@ -542114,6 +542146,189 @@ sethood hypotheses (compare ~ opth ).  (Contributed by BJ, 6-Oct-2018.) $)
+       (Contributed by BJ, 22-Jun-2019.) $)
+    bj-inftyexpidisj $p |- -. ( inftyexpi ` A ) e. CC $=
+      ( vx cinftyexpi cdm wcel cfv cc wn cop wceq cpi opex cvv cnex 0ncn mtbiri
+      c0 syl pm2.61i eleq1d cneg cioc co cv opeq1 df-bj-inftyexpi dmmpti eleq2s
+      fvmpt cpr prid2 csn wo eqid olci wb bj-elopg mpan2 mpbiri bnj1224 sylancr
+      en3lp opprc1 eleq1 eqcom biimpi mtbii ndmfv ) ACDZEZACFZGEZHZVJVKAGIZJZVM
+      VOAKUAKUBUCZVIBABUDZGIZVNVPCVQAGUEBUFZAGLUIBVPVRCVQGLVSUGUHVOVNGEZVLAMEZV
+      THZWAGAGUJZEZWCVNEZWBAGNUKWAWEWCAULJZWCWCJZUMZWGWFWCUNUOWAGMEWEWHUPNAGWCM
+      MUQURUSWDWEVTGWCVNVBUTVAWAHVNQJZWBAGVCWIVTQGEZOVNQGVDPRSVOVNVKGVOVNVKJVKV
+      NVEVFTVGRVJHZVLWJOWKVKQGACVHTPS $.
+  $}
+
+  $( Syntax for the circle at infinity ` CCinfty ` . $)
+  cccinfty $a class CCinfty $.
+
+  $( Definition of the circle at infinity ` CCinfty ` .  (Contributed by BJ,
+     22-Jun-2019.) $)
+  df-bj-ccinfty $a |- CCinfty = ran inftyexpi $.
+
+  ${
+    $d x y $.
+    $( The circle at infinity is disjoint form the set of complex numbers.
+       (Contributed by BJ, 22-Jun-2019.) $)
+    bj-ccinftydisj $p |- ( CC i^i CCinfty ) = (/) $=
+      ( vx vy vz cccinfty cin wcel cinftyexpi cfv wex bj-inftyexpidisj nex wceq
+      cc cv wa elin crn cdm cpi syl wrex wfun cneg cioc df-bj-inftyexpi funmpt2
+      wi co cop elrnrexdm ax-mp rexex df-bj-ccinfty eleq2s anim2i sylbi exancom
+      ancom 19.41v bitri sylbb2 eleq1 biimpac eximi mto bj-nel0 ) AMDEZANZVGFZB
+      NZGHZMFZBIZVLBVJJKVIVHMFZVHVKLZOZBIZVMVIVNVOBIZOZVQVIVNVHDFZOVSVHMDPVTVRV
+      NVRVHGQZDVHWAFZVOBGRZUAZVRGUBWBWDUGCSUCSUDUHCNMUIGCUEUFBGVHUJUKVOBWCULTUM
+      UNUOUPVSVRVNOZVQVNVRURVQVOVNOBIWEVNVOBUQVOVNBUSUTVATVPVLBVOVNVLVHVKMVBVCV
+      DTVEVF $.
+  $}
+
+  $( Syntax for the set of extended complex numbers ` CCbar ` . $)
+  cccbar $a class CCbar $.
+
+  $( Definition of the set of extended complex numbers ` CCbar ` .
+     (Contributed by BJ, 22-Jun-2019.) $)
+  df-bj-ccbar $a |- CCbar = ( CC u. CCinfty ) $.
+
+  $c +cc -cc $.
+
+  $( Syntax for the addition of extended complex numbers. $)
+  caddcc $a class +cc $.
+
+  $( Define the addition of extended complex numbers on the subset of
+     ` ( CCbar X. CCbar ) ` where it makes sense.  (Contributed by BJ,
+     22-Jun-2019.) $)
+  df-bj-addc $a |- +cc =
+    ( x e. ( ( ( CC X. CCbar ) u. ( CCbar X. CC ) ) u. ( Diag ` CCinfty ) ) |->
+        if ( ( 1st ` x ) e. CC , if ( ( 2nd ` x ) e. CC ,
+           ( ( 1st ` x ) + ( 2nd ` x ) ) , ( 2nd ` x ) ) , ( 1st ` x ) ) ) ) $.
+
+  $( Syntax for the opposite of extended complex numbers. $)
+  coppcc $a class -cc $.
+
+  $( Define the opposite of an extended complex number (or more precisely, the
+     operation giving the opposite, sometimes called negation).  (Contributed
+     by BJ, 22-Jun-2019.) $)
+  df-bj-oppc $a |- -cc = ( x e. CCbar |->
+      if ( x e. CC , -u x , ( inftyexpi `
+                          ( if ( 0 < x , ( x - _pi ) , ( x + _pi ) ) ) ) ) ) $.
```

Comment:
I assume you mean ` ( 1st ' x ) ` here instead of ` x `?

Reply:
Bien vu !  commit coming...